File: Beecrowd/6-Paradigmas/1932-Bolsa_de_Valores-7/1932.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){

    int d, t, ax;scanf("%d%d", &d, &t);
    int days[d];
    for(int i =0;i<d;i++){
        scanf("%d", &ax);
        days[i] = ax;
    }
    int ans =0;
    int nextL, currentL;
    currentL = nextL = days[0];
    bool flag = false;
    for(int i=1;i<d;i++){
        if(currentL > days[i] && (currentL - days[i] >= t) || days[i] < nextL ){
            ans = max(ans, ans + currentL - nextL - t);
            currentL = nextL = days[i];
        }
        if(days[i]>currentL){
            currentL = days[i];
        }
    }

    ans = max(ans, ans + currentL - nextL - t);
    
    printf("%d\n", max(0, ans));


    return 0;
}

File: Beecrowd/6-Paradigmas/1932-Bolsa_de_Valores-7/1932.java
import java.util.*;

public class Main {
    public static void main(String[] args){
        int s, t, ant, current, ans;
        Scanner scan = new Scanner(System.in);
        s = scan.nextInt();t = scan.nextInt();
        int[] days = new int[s];
        for(int i =0;i<s;i++)days[i] =  scan.nextInt();
        ans = 0;
        ant = current = days[0];
        for(int i =1;i<s;++i){
            if((current>days[i] && current-days[i]>=t) || ant>days[i]){
                ans = Math.max(ans, ans+current-ant-t);
                current = ant = days[i];
            }
            if(days[i]>current)current=days[i];
        }
        ans=Math.max(ans, ans+current-ant-t);
        System.out.println(ans);

    }
}
